### PR TITLE
Normalise input values for drivetrain

### PR DESCRIPTION
**Pull request by robberwick**
Rather than expect every user of the `DriveTrain` class to use raw pulse lengths, expect input in the range `[-1, 1]` and interpolate internally.

@worlds6440 -  would you take a look at this?

### CODE REVIEW

File: drivetrain.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ def set_neutral(self):
         self.set_servo_pulse(self.channels['right'], self.servo_mid)
 
     # TODO - flesh out setters for raw pulse values (both channels)
-    def mix_channels(self, pulse_throttle, pulse_steering):
+    def mix_channels(self, throttle, steering):
```
Comment (worlds6440) [inline drivetrain.py]: Needs documentation stating that this method mixes throttle and steering but that it also passes those values directly to the motors.

File: drivetrain.py [hunk 2]
```diff
@@ -65,11 +68,11 @@ def mix_channels(self, pulse_throttle, pulse_steering):
         self.set_servo_pulse(self.channels['left'], output_pulse_left)
         self.set_servo_pulse(self.channels['right'], output_pulse_right)
 
-    def map_channel_value(self, value, min_max_range):
+    def _map_channel_value(self, value):
```
Comment (worlds6440) [inline drivetrain.py:78]: Fairly self explanitory, but just needs a one liner comment staying what this method does.

File: rc.py [hunk 3]
```diff
@@ -20,33 +20,19 @@ def run():
...
-                joystick_range
-            )
-
-            drive.mix_channels(pulse_throttle, pulse_steering)
-            logging.info(
-                "stick position: {0}, {1}".format(
-                    acc_throttle, pulse_throttle
-                )
-            )
+            joystick_pos = joystick_state['state']['normalised']
```
Comment (worlds6440) [inline rc.py:35]: Needs short comment to say that joystick_pos is a tuple or throttle and steering.

File: wiimote.py [hunk 4]
```diff
@@ -47,21 +51,31 @@ def __init__(
...
-            joystick_state = self.wm.state['nunchuk']['stick']
-            if clip_values:
-                joystick_state = [
-                    clip(channel, 50, 200)
-                    for channel
-                    in joystick_state
-                ]
-
-            return joystick_state, self.joystick_range
+            joystick_state_raw = self.wm.state['nunchuk']['stick']
```
Comment (worlds6440) [inline wiimote.py:71]: Needs simple one liner comment above the three parts to explain that it stores the raw values, the clipped raw values and the interpolated[-1,1] values.

File: drivetrain.py (hunk 1 above)
Comment (robberwick) [inline drivetrain.py]: Agree that comment is required, perhaps `mix_channels` is also not a descriptive enough method name now. Suggest `mix_channels_and_assign`?

**Comment by robberwick** [comment]
@worlds6440 I've added a bunch of comments/docstrings and updated the `mix_channels` method name. Happy with this?

**Pull request merged**